src/pid_ros.cpp

```diff
@@ -294,40 +312,45 @@ PidROS::setParameterEventCallback()
 
       /// @note don't use getGains, it's rt
       Pid::Gains gains = pid_.getGains();
+      bool changed = false;
 
       for (auto & parameter : parameters) {
         const std::string param_name = parameter.get_name();
         try {
-          if (param_name == topic_prefix_ + "p") {
+          if (param_name == param_prefix_ + "p") {
             gains.p_gain_ = parameter.get_value<double>();
-          } else if (param_name == topic_prefix_ + "i") {
+            changed = true;
+          } else if (param_name == param_prefix_ + "i") {
             gains.i_gain_ = parameter.get_value<double>();
-          } else if (param_name == topic_prefix_ + "d") {
+            changed = true;
+          } else if (param_name == param_prefix_ + "d") {
             gains.d_gain_ = parameter.get_value<double>();
-          } else if (param_name == topic_prefix_ + "i_clamp_max") {
+            changed = true;
+          } else if (param_name == param_prefix_ + "i_clamp_max") {
             gains.i_max_ = parameter.get_value<double>();
-          } else if (param_name == topic_prefix_ + "i_clamp_min") {
+            changed = true;
+          } else if (param_name == param_prefix_ + "i_clamp_min") {
             gains.i_min_ = parameter.get_value<double>();
-          } else if (param_name == topic_prefix_ + "antiwindup") {
+            changed = true;
+          } else if (param_name == param_prefix_ + "antiwindup") {
             gains.antiwindup_ = parameter.get_value<bool>();
-          } else {
-            result.successful = false;
-            result.reason = "Invalid parameter";
+            changed = true;
```

Comment:
Why this change?

Reply:
The node_params_->add_on_set_parameters_callback() adds a callback that will be called whenever any parameter of the node changes. E.g. if you have 2 instances of PidROS, or if you have your own parameters in the same node, then this callback will be called even if a parameter not declared by PidROS changes. If we set result.successful to false because the parameter is not known to PidROS, we are setting false the reconfiguration of the node in general.